src/fparser/tests/test_parser.py

```diff
@@ -334,9 +334,10 @@ def test_nullify():
 
 
 def test_use():
+    ''' Check that we can parse USE statement. '''
     assert parse(Use, 'use a') == 'USE a'
     assert parse(Use, 'use :: a') == 'USE a'
-    assert parse(Use, 'use, intrinsic:: a') == 'USE INTRINSIC :: a'
+    assert parse(Use, 'use, intrinsic:: a') == 'USE, INTRINSIC :: a'
```

Comment:
Please add a test for `use, non_intrinsic :: a` and a case where such a use has an `only` clause

Reply:
Added.